perftest/datanode.go

```
@@ -0,0 +1,108 @@
+package perftest
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	datanode "code.vegaprotocol.io/protos/data-node/api/v1"
+	proto "code.vegaprotocol.io/protos/vega"
+)
+
+func getAssets() error {
+	request := &datanode.AssetsRequest{}
+
+	response, err := dataNode.Assets(context.Background(), request)
+	if err != nil {
+		return err
+	}
+	for _, asset := range response.Assets {
+		assets[asset.Details.Symbol] = asset.Id
+	}
+	return nil
+}
+
+func getAssetsPerUser(pubKey, asset string) (int64, error) {
+	request := &datanode.PartyAccountsRequest{
+		PartyId: pubKey,
+		Asset:   asset,
+	}
+
+	response, err := dataNode.PartyAccounts(context.Background(), request)
+	if err != nil {
+		return 0, err
+	}
+	for _, account := range response.Accounts {
+		return strconv.ParseInt(account.Balance, 10, 64)
+	}
+	return 0, nil
+}
+
+func getMarkets() []string {
+	marketsReq := &datanode.MarketsRequest{}
+
+	response, err := dataNode.Markets(context.Background(), marketsReq)
+	if err != nil {
+		log.Println(err)
+		return nil
+	}
+	marketIDs := []string{}
+	for _, market := range response.Markets {
+		if market.State != proto.Market_STATE_REJECTED {
+			marketIDs = append(marketIDs, market.Id)
+		}
+	}
+	return marketIDs
+}
+
+func getPendingProposalID() (string, error) {
+	request := &datanode.GetProposalsRequest{}
+
+	response, err := dataNode.GetProposals(context.Background(), request)
+	if err != nil {
+		log.Println(err)
```

Might want to return the error here, `response` could be `nil` for all we know, and below would throw a `nil` pointer panic